sdks/java/core/src/main/java/org/apache/beam/sdk/io/LocalFileSystem.java

```diff
@@ -169,11 +183,7 @@ protected void rename(List<LocalResourceId> srcResourceIds, List<LocalResourceId
             dst.getPath());
       }
       // Rename the source file, replacing the existing destination.
-      Files.move(
-          src.getPath(),
-          dst.getPath(),
-          StandardCopyOption.REPLACE_EXISTING,
-          StandardCopyOption.ATOMIC_MOVE);
+      Files.move(src.getPath(), dst.getPath(), renameCopyOption);
```

Comment:
We should probably catch java.nio.file.FileAlreadyExistsException here and raise java.io.FileNotFoundException according to the spec.